src/lcmtype.jl

```diff
@@ -75,13 +157,62 @@ end
     print(io, "This means that you are trying to decode the wrong message type, or a different version of the message type.")
 end
 
-function check_fingerprint(io::IO, ::Type{T}) where T<:LCMType
-    decodefield(io, SVector{8, UInt8}) == fingerprint(T) || throw(FingerprintException(T))
+function checkfingerprint(io::IO, ::Type{T}) where T<:LCMType
+    decodefield(io, Int64) == fingerprint(T) || throw(FingerprintException(T))
+end
+
+# Resizing
+@generated function Base.resize!(x::T) where T<:LCMType
+    exprs = Expr[]
+    for fieldname in fieldnames(T)
+        F = fieldtype(T, fieldname)
+        F <: AbstractVector && push!(exprs, :(LCMCore.resizevec!(x.$fieldname, x, LCMCore.dimensions(T, $(Val(fieldname)))...)))
+    end
+    quote
+        $(exprs...)
+        return nothing
+    end
+end
+
+@inline resizevec!(vec::SVector, x::LCMType, dim::LCMDimension{Int}) = nothing
+@inline resizevec!(vec::Vector, x::LCMType, dim::LCMDimension{Symbol}) = (resize!(vec, getfield(x, dim.size)); nothing)
```

Comment:
Is this `getfield()` access type-unstable?

Reply:
Oh, I see. `dim.size` is a `Val` so it's probably OK.

Reply:
No, this *is* actually type-unstable and `dim.size` is not a `Val`, but the `@inline`s make it not allocate, at least for the examples in the test. I think performance could be improved further by doing even more custom codegen, but it's tricky to avoid calling `dimensions` in the generator (as that would cause world-age issues). So I think this is a pretty good tradeoff.

Reply:
Oh, interesting. If this becomes a problem in the future, could we instead store the actual field name as a type parameter of the dimension (so `LCMDimension{:blength})? Then we could have an `@generated` `getfield` that does the field lookup?

Reply:
Ah yes, that could be a good way out. Yeah, we can do that later if we need even better performance.